Use microsecond precision time for insert/remove

muirmanders opened the pull request.

Commit:
Use microsecond timestamps for thrift inserts.

Previously, same-ms inserts were using the same timestamp which
means the first insert might "win" over the second. Using a new
dependency "microtime", default timestamps now have microsecond
precision which makes things a thousand times better.

The Column class now has a timestamp_micro field you can use if
you want to inspect the high resolution timestamp since javascript
Date objects only support millisecond precision.

Commit:
Make thrift remove use microsecond timestamps.

Like insert, remove now uses microsecond precision timestamps
by default (you can still pass in a Date object).

devdazed commented:
Perfect, thanks!, I still like the idea of gracefully degrading to the original method in case a user doesn't have access to build tools.  I think that if we use [optional dependencies](https://npmjs.org/doc/json.html#optionalDependencies), we can cover all use cases.

Commit:
Move microtime to optionalDependencies.

This way if you have trouble installing/building microtime on Windows, you
can still use helenus (albeit without microsecond precision timestamps).

muirmanders commented:
I took your advice and moved microtime to optionalDependencies.

devdazed commented:
this is included in the 0.6.5 release published today.
